packages/expo-cli/src/commands/build/IOSBuilder.js

```diff
@@ -201,6 +206,8 @@ See https://docs.expo.io/versions/latest/guides/building-standalone-apps.html`
     const publishedExpIds = this.options.publicUrl
       ? undefined
       : await this.ensureReleaseExists('ios');
+
+    await this.checkStatusBeforeBuild({ platform: 'ios', sdkVersion, releaseChannel });
```

Comment:
We should skip calling this function if a user has provided the `--public-url` parameter.